SharpPcap/ICaptureDevice.cs

```diff
@@ -73,7 +73,7 @@ public interface ICaptureDevice : IDisposable
         /// <param name="kernel_buffer_size">
         /// A <see cref="uint"/>
         /// </param>
-        void Open(DeviceModes mode = DeviceModes.None, int read_timeout = 1000, MonitorMode monitor_mode = MonitorMode.Inactive, uint kernel_buffer_size = 0, RemoteAuthentication credentials = null);
+        void Open(DeviceModes mode = DeviceModes.None, int read_timeout = 1000, MonitorMode monitor_mode = MonitorMode.Inactive, uint buffer_size = 0, uint kernel_buffer_size = 0, RemoteAuthentication credentials = null);
```

Comment:
@chmorgan We got 6 arguments here, and if want to fully support all libpcap modes we will need to also add:
* snapshot length https://www.tcpdump.org/manpages/pcap_set_snaplen.3pcap.html
* immediate mode https://www.tcpdump.org/manpages/pcap_set_immediate_mode.3pcap.html
* timestamp type https://www.tcpdump.org/manpages/pcap_set_tstamp_type.3pcap.html
* timestamp precision https://www.tcpdump.org/manpages/pcap_set_tstamp_precision.3pcap.html
* protocol linux https://www.tcpdump.org/manpages/pcap_set_protocol_linux.3pcap.html

We will end up with 10-11 arguments, I think those arguments needs to be changed into a configuration object instead.

Reply:
Yeah it might make sense to go that route now to see how it looks. Are those settings that need to be set prior to adapter activation and not changed at run time? That would seem to be good criteria for a config object.

pcap_set_protocol Linux for example says to use pcap set filter instead which we already provide, so we can skip that API I think.

Don’t we already set the snaplen to the max length?

Reply:
Some uses cases require reducing the snaplen for example:
* you could set the snaplen to 0 that way you only receive the packet length without data, and create statistics from it.
* you could set the snaplen small enough to include only the IP address, this way you can create a small pcap file that acts as an access log file.